src/library_addfunction.js

```diff
@@ -23,12 +23,7 @@ mergeInto(LibraryManager.library, {
   $sigToWasmTypes: function(sig) {
     var typeNames = {
       'i': 'i32',
-#if MEMORY64
       'j': 'i64',
-#else
-      // i64 values will be split into two i32s.
```

Comment:
Don't you still need this case for then WASM_BIGINT is not set?  Should we instead replace `MEMORY64` with `WASM_BIGINT`?

Reply:
I see.  I think what might be happening is that case is that we will be using BigInt integration, despite the fact that WASM_BIGINT is not set.   What happens if you try to run this test on an ancient version of node (for example the oldest version of node we support which is 10.19.0 (see test-node-compat once again for details of how we download and configure that in CI).

Reply:
I think the way it works is that it's on the caller to know about this? Because if we split the arguments, the function we are adding needs to know about it. When WASM_BIGINT isn't available, the runtime needs to make a different version of the function we are adding that knows about the no-wasm-bigint abi and handles that correctly. Then it should replace the `j` with `ii` etc.

Reply:
Unless we want to change `addFunction` to also generate the adapter.

>  I think what might be happening is that case is that we will be using BigInt integration, despite the fact that WASM_BIGINT is not set.

Yes, you must be right about this.

Reply:
I think that best option here would probably be to assert if `j` is used without `WASM_BIGINT` support.  Something like  `assert('j' not in sig, "i64 not permitted in function signatures when WASM_BIGINT is disabled)`

Reply:
Ah wait. `WebAssembly.Function` existing implies wasm bigint integration is available as a browser feature. So we will never be on this code path without wasm bigint support.

Reply:
Okay I added an assertion as you suggested.